### ISSUE TITLE
val_check_interval equivalent for training loss logging

### ISSUE THREAD

**Issue by coloss** [PyTorchLightning/pytorch-lightning#6180]
I very much like the feature of `val_check_interval`. I use it for logging the validation loss when the epochs are too large. However, I would like to log the training loss at the same steps as well so that for every validation loss entry in my logs, I also have a corresponding training loss entry. 
Is there an easy way to do it, or would I have to write my own logger for that? 

Thank you.

**Comment by s-rog** [comment]
If you are logging manually in `training_epoch_end` then you can simply use something like `if self.trainer.current_epoch % n ==0`

This is the simplest method off the top of my head.

**Comment by tchaton** [comment]
Dear @coloss,

What would happen if you set `Trainer(val_check_interval=50, log_every_n_steps=50)` ?

Best,
T.C

**Comment by coloss** [comment]
> If you are logging manually in `training_epoch_end` then you can simply use something like `if self.trainer.current_epoch % n ==0`
> 
> This is the simplest method off the top of my head.

I actually use the version where I specify a float, so I could validate in "sub-epochs". I guess I could still do what you say, but put the condition into `training_step_end` and just log whenever the desired fraction of the entire epoch is process. Somehow it feels a little hacky though.

**Comment by coloss** [comment]
> Dear @coloss,
> 
> What would happen if you set `Trainer(val_check_interval=50, log_every_n_steps=50)` ?
> 
> Best,
> T.C

thanks, @tchaton , Looking at the implementation,  `log_every_n_steps` is set to 50 by default if you don't specify otherwise. I don't believe this parameter helps me to log training loss for a fraction of an epoch.

**Comment by s-rog** [comment]
If you're using pl metrics you can also add data points in the training step, manually call compute at custom step intervals and log the output.
